server/src/test/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessorTest.java

```diff
@@ -322,6 +325,68 @@ public void alignedTvListRamCostTest()
     Assert.assertEquals(721560, memTable.memSize());
   }
 
+  @Test
+  public void testMemTableMapRamCost()
+      throws MetadataException, WriteProcessException, IOException {
+    processor =
+        new TsFileProcessor(
+            storageGroup,
+            SystemFileFactory.INSTANCE.getFile(filePath),
+            sgInfo,
+            this::closeTsFileProcessor,
+            (tsFileProcessor) -> true,
+            true);
+    TsFileProcessorInfo tsFileProcessorInfo = new TsFileProcessorInfo(sgInfo);
+    processor.setTsFileProcessorInfo(tsFileProcessorInfo);
+    this.sgInfo.initTsFileProcessorInfo(processor);
+    SystemInfo.getInstance().reportStorageGroupStatus(sgInfo, processor);
+    processor.insertTablet(genInsertTablePlan(0, false), 0, 10, new TSStatus[10]);
+    IMemTable memTable = processor.getWorkMemTable();
+    long memTableMapRamCost = memTable.getMemTableMapRamCost();
+    processor.insertTablet(genInsertTablePlan(1001, false), 0, 10, new TSStatus[10]);
+    Assert.assertEquals(memTableMapRamCost, memTable.getMemTableMapRamCost());
+
+    TSRecord record = new TSRecord(123L, "root.sg.device5");
+    record.addTuple(DataPoint.getDataPoint(dataType, "ns0", String.valueOf(1)));
+    processor.insert(new InsertRowPlan(record));
+    Assert.assertEquals(
+        memTableMapRamCost
+            + RamUsageEstimator.shallowSizeOfInstance(WritableMemChunk.class)
+            + RamUsageEstimator.sizeOf("ns0"),
+        memTable.getMemTableMapRamCost());
+  }
+
+  @Test
+  public void testAlignedMemTableMapRamCost()
+      throws MetadataException, WriteProcessException, IOException {
+    processor =
+        new TsFileProcessor(
+            storageGroup,
+            SystemFileFactory.INSTANCE.getFile(filePath),
+            sgInfo,
+            this::closeTsFileProcessor,
+            (tsFileProcessor) -> true,
+            true);
+    TsFileProcessorInfo tsFileProcessorInfo = new TsFileProcessorInfo(sgInfo);
+    processor.setTsFileProcessorInfo(tsFileProcessorInfo);
+    this.sgInfo.initTsFileProcessorInfo(processor);
+    SystemInfo.getInstance().reportStorageGroupStatus(sgInfo, processor);
+    processor.insertTablet(genInsertTablePlan(0, true), 0, 10, new TSStatus[10]);
+    IMemTable memTable = processor.getWorkMemTable();
+    long memTableMapRamCost = memTable.getMemTableMapRamCost();
+    processor.insertTablet(genInsertTablePlan(1001, true), 0, 10, new TSStatus[10]);
+    Assert.assertEquals(memTableMapRamCost, memTable.getMemTableMapRamCost());
+
+    TSRecord record = new TSRecord(123L, "root.sg.device5");
+    record.addTuple(DataPoint.getDataPoint(dataType, "ns0", String.valueOf(1)));
+    processor.insert(new InsertRowPlan(record));
```

Comment:
```suggestion
    InsertRowPlan insertRowPlan = new InsertRowPlan(record);
    insertRowPlan.setAligned(true);
    processor.insert(insertRowPlan);
```

Reply:
This plan has been set to aligned. And accuracy check has been added to ut.